File: 2610-convert-an-array-into-a-2d-array-with-conditions/2610-convert-an-array-into-a-2d-array-with-conditions.cpp
class Solution {
public:
    vector<vector<int>> findMatrix(vector<int>& nums) {
        unordered_map<int,int> mp;
        int n=0;
        for(auto i: nums) mp[i]++, n= max(n, mp[i]);
        vector<vector<int>> ans(n);
        for(auto i: mp){
            for(int j=0; j<i.second; j++){
                ans[j].push_back(i.first);
            }
        }
        return ans;
    }
};

File: 2610-convert-an-array-into-a-2d-array-with-conditions/2610-convert-an-array-into-a-2d-array-with-conditions.java
class Solution {
    public List<List<Integer>> findMatrix(int[] nums) {
        Arrays.sort(nums);
        List<List<Integer>> res = new ArrayList<>();
        int idx=-1, num=nums[0];
        for(int i=0; i<nums.length; i++){
            if(nums[i]== num){
                if(res.size()==idx+1){
                    res.add(new ArrayList<>());
                }
                idx++;
            }else{
                num=nums[i];
                idx=0;
            }
            res.get(idx).add(nums[i]);
        }
        return res;
    }
}